helpers/View.php

```diff
@@ -139,33 +159,49 @@ public static function getGlobalJsFileName() {
      * @return string all.css file name
      */
     public static function getGlobalCssFileName() {
-        return 'all-v' . \F3::get('version') . '.css';
+        return self::$staticPrefix . self::$staticmtime['css'] . '.css';
     }
 
     /**
      * generate minified css and js
      *
+     * @param type js or css
+     *
      * @return void
      */
-    public function genMinifiedJsAndCss() {
-        // minify js
-        $targetJs = \F3::get('BASEDIR') . '/public/' . self::getGlobalJsFileName();
-        if (!file_exists($targetJs) || \F3::get('DEBUG') != 0) {
-            $js = '';
-            foreach (\F3::get('js') as $file) {
-                $js = $js . "\n" . $this->minifyJs(file_get_contents(\F3::get('BASEDIR') . '/' . $file));
+    private function genMinified($type) {
+        self::$staticmtime[$type] = self::maxmtime(\F3::get($type));
+
+        if ($type == 'js') {
+            $filename = self::getGlobalJsFileName();
+        } elseif ($type == 'css') {
+            $filename = self::getGlobalCssFileName();
+        }
+        $target = \F3::get('BASEDIR') . '/public/' . $filename;
+
+        // build if needed
+        $builtSomething = false;
+        if (!file_exists($target)) {
+            $minified = '';
+            foreach (\F3::get($type) as $file) {
+                if ($type == 'js') {
+                    $minifiedFile = $this->minifyJs(file_get_contents(\F3::get('BASEDIR') . '/' . $file));
+                } elseif ($type == 'css') {
+                    $minifiedFile = $this->minifyCss(file_get_contents(\F3::get('BASEDIR') . '/' . $file));
+                }
+                $minified = $minified . "\n" . $minifiedFile;
             }
-            file_put_contents($targetJs, $js);
+            file_put_contents($target, $minified);
+            $builtSomething = true;
         }
 
-        // minify css
-        $targetCss = \F3::get('BASEDIR') . '/public/' . self::getGlobalCssFileName();
-        if (!file_exists($targetCss) || \F3::get('DEBUG') != 0) {
-            $css = '';
-            foreach (\F3::get('css') as $file) {
-                $css = $css . "\n" . $this->minifyCss(file_get_contents(\F3::get('BASEDIR') . '/' . $file));
+        // cleanup
+        if ($builtSomething) {
```

Comment:
One disadvantage of cache busting using a query string is that it does not return a 404 for outdated content. But is saves the cleanup. Anyway, both may lower user support needs.